From 20ff2ee50912aaade64ffcda1acdc3b1f129a05f Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:38:32 -0400
Subject: [ethernet] i40e: Fix incorrect feature configuration status

Message-id: <1409658042-22689-152-git-send-email-sassmann@redhat.com>
Patchwork-id: 90348
O-Subject: [RHEL7.1 PATCH 151/281] i40e: Fix incorrect feature configuration status
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Akeem G Abodunrin <akeem.g.abodunrin@intel.com>

This patch fixes an issue where FD SB/ATR and NTUPLE configurations status are
reported erroneously. Without this patch, driver reports FDir without further
information.

Change-ID: I5bdd2871b7f2db1e5f5e76c741ae6a0dc603b453
Signed-off-by: Akeem G Abodunrin <akeem.g.abodunrin@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit c6423ff1005902dd17f2efca5f1a0716e5259283)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 9b85b2d..a3c5ce4 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -8237,11 +8237,12 @@ static void i40e_print_features(struct i40e_pf *pf)
 
  if (pf->flags & I40E_FLAG_RSS_ENABLED)
   buf += sprintf(buf, "RSS ");
- buf += sprintf(buf, "FDir ");
  if (pf->flags & I40E_FLAG_FD_ATR_ENABLED)
-  buf += sprintf(buf, "ATR ");
- if (pf->flags & I40E_FLAG_FD_SB_ENABLED)
+  buf += sprintf(buf, "FD_ATR ");
+ if (pf->flags & I40E_FLAG_FD_SB_ENABLED) {
+  buf += sprintf(buf, "FD_SB ");
   buf += sprintf(buf, "NTUPLE ");
+ }
  if (pf->flags & I40E_FLAG_DCB_ENABLED)
   buf += sprintf(buf, "DCB ");
  if (pf->flags & I40E_FLAG_PTP)
-- 
1.7.1